.github/workflows/binaries-and-deb-release.yml

```
@@ -0,0 +1,618 @@
+              raise SystemExit(f"Missing binary at {binary_path}")
+          checksum = hashlib.sha256(binary_path.read_bytes()).hexdigest()
+          output = binary_path.parent / f"{binary_path.name}.sbom.spdx"
+          doc = f"""SPDXVersion: SPDX-2.3
+          DataLicense: CC0-1.0
+          SPDXID: SPDXRef-DOCUMENT
+          DocumentName: {binary_path.name}
+          DocumentNamespace: https://github.com/{os.environ['GITHUB_REPOSITORY']}/sbom/{binary_path.name}-{checksum}
+          Creator: Tool: zallet-ci-spdx-1.0
+          Created: 1970-01-01T00:00:00Z
+          PackageName: {binary_path.name}
+          PackageSPDXID: SPDXRef-Package-{checksum[:12]}
+          PackageDownloadLocation: NOASSERTION
+          FilesAnalyzed: false
+          PackageChecksum: SHA256: {checksum}
+          """
+          output.write_text(doc, encoding="utf-8")
+          PY
+
+      - name: Generate build provenance attestation for binary
+        id: binary-attest
+        uses: actions/attest-build-provenance@977bb373ede98d70efdf65b84cb5f73e068dcc2a # v3.0.0
+        with:
+          subject-path: standalone/zallet-${{ env.RELEASE_VERSION }}-${{ matrix.binary_suffix }}
+
+      - name: Save binary attestation bundle
+        env:
+          BUNDLE_PATH: ${{ steps.binary-attest.outputs['bundle-path'] }}
+        run: |
+          cp -- "$BUNDLE_PATH" "standalone/zallet-${RELEASE_VERSION}-${{ matrix.binary_suffix }}.intoto.jsonl"
+
+      - name: Seed cargo target tree with the imported artifacts
+        run: |
+          set -euo pipefail
+
+          PLATFORM_KEY="${{ matrix.platform_key }}"
+          TARGET_TRIPLE="${{ matrix.target }}"
+
+          # Root where the export stage stored all Zallet assets under /usr/local/share/zallet
+          ROOT="build/${PLATFORM_KEY}/usr/local/share/zallet"
+
+          TARGET_DIR="target/${TARGET_TRIPLE}/release"
+          COMPLETIONS_DIR="${TARGET_DIR}/completions"
+          MANPAGES_DIR="${TARGET_DIR}/manpages"
+
+          mkdir -p "${TARGET_DIR}" "${COMPLETIONS_DIR}" "${MANPAGES_DIR}"
+
+          # Seed binary for cargo-deb --no-build
+          install -m 0755 "build/${PLATFORM_KEY}/zallet" "${TARGET_DIR}/zallet"
+
+          # Copy the whole completions tree (whatever is inside)
+          cp -a "${ROOT}/completions/." "${COMPLETIONS_DIR}/" 2>/dev/null || true
+
+          # Copy the whole manpages tree
+          cp -a "${ROOT}/manpages/." "${MANPAGES_DIR}/" 2>/dev/null || true
+
+          # Copy Debian copyright metadata
+          install -m 0644 "${ROOT}/debian-copyright" \
+            "${TARGET_DIR}/debian-copyright" 2>/dev/null || true
+
+
+      - name: Build package (cargo deb --no-build)
+        run: |
+          cargo deb --no-build --target ${{ matrix.target }}
+
+      - name: Rename package to not collide across the matrix
+        run: |
+          mv target/debian/zallet_*.deb ./
+          mv zallet_*.deb "zallet_${RELEASE_VERSION}_${{ matrix.arch }}.deb"
```

Bug: this re-introduces an earlier bug we had with the old `deb-release.yml` (fixed in 7ad338b3a1d9bd2d7d6cde70d9f1d56c453ec8ca) in that it does not include `${{ matrix.debian }}`, so builds for different Debian distros but the same arch will collide.

Either use the previous Bash logic we had in `deb-release.yml`:
```suggestion
          mv zallet_*.deb "$(basename -s .deb zallet_*.deb)_${{ matrix.debian }}.deb"
```
or explicitly include the Debian version:
```suggestion
          mv zallet_*.deb "zallet_${RELEASE_VERSION}_${{ matrix.arch }}_${{ matrix.debian }}.deb"
```

In either case, every reference to the .deb file in subsequent steps needs to be fixed to account for this extra suffix (like we also did in `deb-release.yml`).